mysql-test/suite/rpl/t/rpl_rewrite_db_sys_vars.test

```diff
@@ -97,6 +101,28 @@ SELECT @@GLOBAL.replicate_rewrite_db;
 # This shouldn't get the new values from x DB on master
 select * from y.my_table;
 
+--echo # Dynamic updates to the replication filter should be lost after server restart
+# Old value
+connection slave;
+SELECT @@GLOBAL.replicate_rewrite_db;
+
+connection master;
+use x;
+insert into my_table values (1000);
+select * from my_table;
+--source include/save_master_gtid.inc
+
+--let $rpl_server_number= $rpl_server_id
+--source include/rpl_restart_server.inc
+
+# New value
+connection slave;
+--source include/start_slave.inc
+SELECT @@GLOBAL.replicate_rewrite_db;
+--source include/sync_with_master_gtid.inc
+# This should update values with 314 and 1000 from primary (BUT IS NOT !~)
```

Comment:
Right. The slave's GTID position is already caught up with the master at the time of `rpl_restart_server.inc`. To have the events replayed with the correct filter, reset the slave before the call to `rpl_restart_server.inc`:
```
--source include/stop_slave.inc
--source include/reset_slave.inc
--source include/start_slave.inc
--let $rpl_server_number= $rpl_server_id
--source include/rpl_restart_server.inc
```